src/main/java/com/drajer/bsa/ehr/service/impl/EhrFhirR4QueryServiceImpl.java

```diff
@@ -130,6 +121,41 @@
     return kd.getFhirInputData();
   }
 
+  public IGenericClient getClient(KarProcessingData kd, FhirContext context) {
+    String secret = kd.getHealthcareSetting().getClientSecret();
+    if (secret == null || secret.isEmpty()) {
+      backendAuthorizationService.getAuthorizationToken(kd);
+    } else {
+      ehrAuthorizationService.getAuthorizationToken(kd);
+    }
+
+    return fhirContextInitializer.createClient(
+        context,
+        kd.getHealthcareSetting().getFhirServerBaseURL(),
+        kd.getNotificationContext().getEhrAccessToken());
+  }
+
+  public void createResource(KarProcessingData kd, Resource resource) {
+
+    logger.info(" Getting FHIR Context for R4");
+    FhirContext context = fhirContextInitializer.getFhirContext(R4);
+
+    logger.info("Initializing FHIR Client");
+    IGenericClient client = getClient(kd, context);
+    client.create().resource(resource);
```

Comment:
I may be missing something but when I tested this as-is it didn't work, there was no activity on the EHR server. I think we need to add an `execute` call to the end of this so that it actually performs the intended action:
```suggestion
    client.create().resource(resource).execute();
```